platform/mbed_watchdog_mgr.h

```diff
@@ -0,0 +1,97 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2018 ARM Limited
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef MBED_WATCHDOG_MGR_H
+#define MBED_WATCHDOG_MGR_H
+
+#ifdef DEVICE_WATCHDOG
+
+#include "watchdog_api.h"
+#include "mbed_error.h"
+#include "platform/Callback.h"
+#include "platform/mbed_critical.h"
+#include "platform/SingletonPtr.h"
+#if DEVICE_LPTICKER
+#include "LowPowerTicker.h"
+#else
+#include "Ticker.h"
+#endif
+#include "Watchdog.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** \addtogroup platform */
+/** A system timer that will reset the system in the case of system failures or
+ *  malfunctions.
+ *
+ * Example:
```

Comment:
We should have the example showing how to use it and not how to hang and reset the system.